### PR TITLE
feat(archetype): add archetype for rest service delete

### PR DESCRIPTION
**Pull request by eaquin02**
This PR is for adding the synapse-archetype-service-rest-delete module.

### CODE REVIEW

**Review by whoswendy — CHANGES_REQUESTED** [review]
Please add dependency to main pom.xml and add a __className__Application.java

**Review by whoswendy — APPROVED** [review]

File: archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/main/java/service/__className__DeleteService.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package ${package}.service;
```
Comment (whoswendy) [inline archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/main/java/service/__className__DeleteService.java:14]: please add copyright

**Review by whoswendy — CHANGES_REQUESTED** [review]
please add missing copyright

**Review by whoswendy — APPROVED** [review]

File: archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/main/java/config/__className__DeleteServiceEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.PropertySource;
+
+/**
+ * {@code ${className}DeleteServiceEndpoint} class sets the endpoints
+ * for the {@link ${className}DeleteService}.
+ * @author ${author}
+ */
+public class ${className}DeleteServiceEndpoint {
+    /**
```
Comment (guttff) [inline archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/main/java/config/__className__DeleteServiceEndpoint.java:30]: add a private constructor

File: archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/test/java/service/__className__DeleteServiceIT.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ */
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes=${className}DeleteServiceTestConfig.class)
+class ${className}DeleteServiceIT {
+
+    @Autowired
+    private ${className}DeleteService restService;
+
+    @Test
+    void callDeleteService_givenValidRequest_expectedValidResponse() throws Exception {
```
Comment (guttff) [inline archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/test/java/service/__className__DeleteServiceIT.java]:
```suggestion
    void delete_givenValidRequest_expectedValidResponse() throws Exception {
```

Comment (guttff) [reply archetype/synapse-archetype-service-rest-delete/src/main/resources/archetype-resources/src/test/java/service/__className__DeleteServiceIT.java]: We should keep the name of the method exactly how the actual method is named

**Review by guttff — APPROVED** [review]

**Pull request merged**